neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/data/im/InMemoryBackendFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2013-2017 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.data.im;
+
+import fr.inria.atlanmod.commons.annotation.Static;
+import fr.inria.atlanmod.neoemf.config.ImmutableConfig;
+import fr.inria.atlanmod.neoemf.data.AbstractBackendFactory;
+import fr.inria.atlanmod.neoemf.data.Backend;
+import fr.inria.atlanmod.neoemf.data.BackendFactory;
+
+import org.eclipse.emf.common.util.URI;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * A factory that create {@link DefaultInMemoryBackend} instances.
+ */
+@ParametersAreNonnullByDefault
+public class InMemoryBackendFactory extends AbstractBackendFactory {
+
+    /**
+     * The literal description of the factory.
+     */
+    public static final String NAME = "im"; // In-Memory
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename field "NAME" to prevent any misunderstanding/clash with method "name" defined on line 51. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1845)